core/src/main/java/org/elasticsearch/rest/action/search/RestMultiSearchAction.java

```diff
@@ -96,7 +100,11 @@ public static MultiSearchRequest parseRequest(RestRequest restRequest, boolean a
                 throw new ElasticsearchParseException("Exception when parsing search request", e);
             }
         });
-
+        List<SearchRequest> requests = multiRequest.requests();
+        preFilterShardsAfter = Math.max(1, preFilterShardsAfter / (requests.size()+1));
+        for (SearchRequest request : requests) {
+            request.setPreFilterSearchShardsAfter(preFilterShardsAfter);
```

Comment:
Should we check if `preFilterShardsAfter` has been set explicitly on the search request and set it to the min of `preFilterShardsAfter` and `request.getPreFilterSearchShardsAfter()`? Not sure if this would actually matter in practice?